### ISSUE TITLE
ethminer stops reporting hashrate to the pool after couple hours

### ISSUE THREAD

**Issue by akatasonov** [ethereum-mining/ethminer#888]
Running ethminer version 0.14.0.dev3+git.de05cc3
Build: linux/release/gnu

the -RH option is specified and the hashrate reporting works for a while but then just stops working. Please see the image below from ethermine

![image](https://user-images.githubusercontent.com/6917547/37359473-5a57e430-26f6-11e8-9d80-0f25e338493e.png)

**Comment by smurfy** [comment]
please try dev4, also what does the console say? are you seeing stats there?

**Comment by radus79** [comment]
Same issue on 0.14.0 dev4 running on Windows. Hashrate reporting stops after 20-30 minutes. On console everything is OK. Average hashrate on pool side looks good too.

**Comment by akatasonov** [comment]
@smurfy console shows happy mining, no error messages whatsoever

**Comment by smurfy** [comment]
hm dev4 changed the way to report a bit. need to do more tests with it.

**Comment by akatasonov** [comment]
@smurfy  My version of ethminer is 0.14.0.dev3 so apparently the problem is not caused by changes introduced in dev4

**Comment by kalltkaffe** [comment]
Seeing same thing in Linux aswell. Everything else works, average and current hashrates are normal and console reports same hashrate as usual. It's just on the the ethermine stats that the reported hashrate suddenly is missing.

**Comment by sandrodz** [comment]
I can confirm this bug, 0.14.0.dev4 on linux.

<img width="221" alt="screen shot 2018-03-14 at 12 37 29 am" src="https://user-images.githubusercontent.com/8479569/37368498-e8e491a2-271f-11e8-98c4-b0bab3d541a3.png">

No errors in the log.

**Comment by propeller411** [comment]
I hav this problem too, 0.14.0.dev4, windows 10 C:\bin\ethminer.exe -RH -HWMON 1 -U -S 188.165.226.213:5555 -FS 94.23.36.128:5555 -O my_wallet.my_worker --stratum-ssl --cuda-parallel-hash 4
0.14.0.dev3, windows 10 is OK
![default](https://user-images.githubusercontent.com/26086350/37369724-008edcf4-271b-11e8-8ef2-6be473027a12.png)

**Comment by propeller411** [comment]
Return to 0.14.0.dev3 solves the problem

**Comment by kalltkaffe** [comment]
I suppose it might be this commit that cause the problem? It seems to change the code used to submit hashrate: https://github.com/ethereum-mining/ethminer/commit/209254b6990119853d14d517bb3b315813cab062

**Comment by jmsjr** [comment]
I have this problem too ... interestingly, I have 2 instances of ethminer running ( one for OpenCL and one for CUDA ). When the instance for OpenCL segfaulted after disconnecting due to another issue, BOTH instances / processes are not reporting their hashrate anymore, even the process / instance running for CUDA that is still running.

**Comment by akatasonov** [comment]
dev4 has this issue as well as dev3 (just upgraded from dev3 to dev4)

![image](https://user-images.githubusercontent.com/6917547/37426155-45024c46-27ce-11e8-8452-e22efe0ab157.png)

**Comment by jmsjr** [comment]
I don't think dev3 has the problem. Your version says:

```
Running ethminer version 0.14.0.dev3+git.de05cc3
Build: linux/release/gnu
```

.. but the official dev03 release ( at least the one that I have that is NOT built from git master ) says:

```
ethminer version 0.14.0.dev3+git.83693003
```

**Comment by jean-m-cyr** [comment]
> I suppose it might be this commit that cause the problem? It seems to change the code used to submit hashrate: 209254b

Could be... although that was introduced to address another issue, which was concurrent use of a common SSL socket by two independent threads.

**Comment by kalltkaffe** [comment]
> Could be... although that was introduced to address another issue, which was concurrent use of a common SSL socket by two independent threads.

I manually changed the code back, recompiled and then restarted mining. Time will tell if it helped.

**Comment by jean-m-cyr** [comment]
This is duplicate of issue #824, opened before intro of 209254b

**Comment by jmsjr** [comment]
I think this is related to https://github.com/ethereum-mining/ethminer/issues/890

1) I have two instances of ethminer running last night, one with SSL / TLS  and another  without using SSL / TLS ... each instance using a different set of GPUs.
2) The instance with SSL / TLS segfaulted ( due to the issue outlined in https://github.com/ethereum-mining/ethminer/issues/890 .. which is a continuation of an earlier issue )
3) At the same time when that happened, the instance of ethminer without using SSL / TLS did NOT segfault, but after that, the stats for this worker on ethminer.org are gone ( e.g. ethminer stopped reporting the hashrate ).
4) What happened at that time was that, according to dmesg and /var/log/syslog ... my internet connection went "bad" and wireless network interface was restarted. See first line below about the segfault ( which is the segfault for the instance of ethminer WITH SSL )

```
Mar 15 02:17:52 rig01 kernel: [102567.452274] stratum[9266]: segfault at 28e9560 ip 00000000028e9560 sp 00007f76f731ab48 error 15
Mar 15 02:17:52 rig01 systemd-timesyncd[791]: Timed out waiting for reply from 198.18.1.1:123 (ntp.ubuntu.com).
Mar 15 02:20:55 rig01 wpa_supplicant[1112]: wlx001ee5fc8a0a: WPA: Group rekeying completed with e0:b9:e5:99:02:43 [GTK=CCMP]
Mar 15 02:23:15 rig01 dhclient[1230]: DHCPREQUEST of 10.1.1.51 on wlx001ee5fc8a0a to 10.1.1.1 port 67 (xid=0x52d0e195)
Mar 15 02:23:15 rig01 dhclient[1230]: DHCPACK of 10.1.1.51 from 10.1.1.1
Mar 15 02:23:15 rig01 NetworkManager[972]: <info>  [1521040995.6894]   address 10.1.1.51
Mar 15 02:23:15 rig01 NetworkManager[972]: <info>  [1521040995.6895]   plen 24 (255.255.255.0)
Mar 15 02:23:15 rig01 NetworkManager[972]: <info>  [1521040995.6895]   gateway 10.1.1.1
Mar 15 02:23:15 rig01 NetworkManager[972]: <info>  [1521040995.6895]   server identifier 10.1.1.1
Mar 15 02:23:15 rig01 NetworkManager[972]: <info>  [1521040995.6896]   lease time 86400
Mar 15 02:23:15 rig01 NetworkManager[972]: <info>  [1521040995.6896]   hostname 'rig01'
Mar 15 02:23:15 rig01 NetworkManager[972]: <info>  [1521040995.6896]   nameserver '10.1.1.1'
Mar 15 02:23:15 rig01 NetworkManager[972]: <info>  [1521040995.6897]   domain name 'lan'
Mar 15 02:23:15 rig01 NetworkManager[972]: <info>  [1521040995.6897] dhcp4 (wlx001ee5fc8a0a): state changed bound -> bound
Mar 15 02:23:15 rig01 dbus[951]: [system] Activating via systemd: service name='org.freedesktop.nm_dispatcher' unit='dbus-org.freedesktop.nm-dispatcher.service
'
Mar 15 02:23:15 rig01 systemd[1]: Starting Network Manager Script Dispatcher Service...
Mar 15 02:23:15 rig01 dhclient[1230]: bound to 10.1.1.51 -- renewal in 38911 seconds.
Mar 15 02:23:15 rig01 dbus[951]: [system] Successfully activated service 'org.freedesktop.nm_dispatcher'
Mar 15 02:23:15 rig01 systemd[1]: Started Network Manager Script Dispatcher Service.
Mar 15 02:23:15 rig01 nm-dispatcher: req:1 'dhcp4-change' [wlx001ee5fc8a0a]: new request (1 scripts)
Mar 15 02:23:15 rig01 nm-dispatcher: req:1 'dhcp4-change' [wlx001ee5fc8a0a]: start running ordered scripts...
Mar 15 02:30:55 rig01 wpa_supplicant[1112]: wlx001ee5fc8a0a: WPA: Group rekeying completed with e0:b9:e5:99:02:43 [GTK=CCMP]
Mar 15 02:40:55 rig01 wpa_supplicant[1112]: wlx001ee5fc8a0a: WPA: Group rekeying completed with e0:b9:e5:99:02:43 [GTK=CCMP]
Mar 15 02:44:14 rig01 wpa_supplicant[1112]: wlx001ee5fc8a0a: CTRL-EVENT-DISCONNECTED bssid=e0:b9:e5:99:02:43 reason=4 locally_generated=1
Mar 15 02:44:14 rig01 NetworkManager[972]: <warn>  [1521042254.0153] sup-iface[0x14bf2d0,wlx001ee5fc8a0a]: connection disconnected (reason -4)
Mar 15 02:44:14 rig01 wpa_supplicant[1112]: wlx001ee5fc8a0a: CTRL-EVENT-REGDOM-CHANGE init=CORE type=WORLD
Mar 15 02:44:14 rig01 NetworkManager[972]: <info>  [1521042254.0388] device (wlx001ee5fc8a0a): supplicant interface state: completed -> disconnected
Mar 15 02:44:14 rig01 NetworkManager[972]: <info>  [1521042254.1233] device (wlx001ee5fc8a0a): supplicant interface state: disconnected -> scanning
Mar 15 02:44:15 rig01 wpa_supplicant[1112]: wlx001ee5fc8a0a: SME: Trying to authenticate with e0:b9:e5:99:02:43 (SSID='iiNet990243' freq=2412 MHz)
Mar 15 02:44:15 rig01 kernel: [104150.349353] wlx001ee5fc8a0a: authenticate with e0:b9:e5:99:02:43
Mar 15 02:44:15 rig01 NetworkManager[972]: <info>  [1521042255.3735] device (wlx001ee5fc8a0a): supplicant interface state: scanning -> authenticating
Mar 15 02:44:15 rig01 kernel: [104150.389916] wlx001ee5fc8a0a: send auth to e0:b9:e5:99:02:43 (try 1/3)
Mar 15 02:44:15 rig01 kernel: [104150.592451] wlx001ee5fc8a0a: send auth to e0:b9:e5:99:02:43 (try 2/3)
Mar 15 02:44:15 rig01 kernel: [104150.796449] wlx001ee5fc8a0a: send auth to e0:b9:e5:99:02:43 (try 3/3)
Mar 15 02:44:15 rig01 kernel: [104151.000449] wlx001ee5fc8a0a: authentication with e0:b9:e5:99:02:43 timed out
Mar 15 02:44:16 rig01 NetworkManager[972]: <info>  [1521042256.0253] device (wlx001ee5fc8a0a): supplicant interface state: authenticating -> disconnected
Mar 15 02:44:16 rig01 NetworkManager[972]: <info>  [1521042256.5285] device (wlx001ee5fc8a0a): supplicant interface state: disconnected -> scanning
Mar 15 02:44:29 rig01 NetworkManager[972]: <warn>  [1521042269.5804] device (wlx001ee5fc8a0a): link timed out.
Mar 15 02:44:29 rig01 NetworkManager[972]: <info>  [1521042269.5804] device (wlx001ee5fc8a0a): state change: activated -> failed (reason 'ssid-not-found') [100 120 53]
Mar 15 02:44:29 rig01 NetworkManager[972]: <info>  [1521042269.5810] manager: NetworkManager state is now CONNECTED_LOCAL
Mar 15 02:44:29 rig01 NetworkManager[972]: <info>  [1521042269.5813] manager: NetworkManager state is now DISCONNECTED
Mar 15 02:44:29 rig01 whoopsie[1121]: [02:44:29] offline
Mar 15 02:44:29 rig01 NetworkManager[972]: <warn>  [1521042269.5867] device (wlx001ee5fc8a0a): Activation: failed for connection 'iiNet990243'
Mar 15 02:44:29 rig01 dbus[951]: [system] Activating via systemd: service name='org.freedesktop.nm_dispatcher' unit='dbus-org.freedesktop.nm-dispatcher.service'
Mar 15 02:44:29 rig01 NetworkManager[972]: <info>  [1521042269.5896] device (wlx001ee5fc8a0a): state change: failed -> disconnected (reason 'none') [120 30 0]
Mar 15 02:44:29 rig01 avahi-daemon[925]: Withdrawing address record for fe80::a37f:5727:c092:929d on wlx001ee5fc8a0a.
Mar 15 02:44:29 rig01 avahi-daemon[925]: Leaving mDNS multicast group on interface wlx001ee5fc8a0a.IPv6 with address fe80::a37f:5727:c092:929d.
Mar 15 02:44:29 rig01 avahi-daemon[925]: Interface wlx001ee5fc8a0a.IPv6 no longer relevant for mDNS.
Mar 15 02:44:29 rig01 systemd[1]: Starting Network Manager Script Dispatcher Service...
Mar 15 02:44:29 rig01 kernel: [104164.608918] IPv6: ADDRCONF(NETDEV_UP): wlx001ee5fc8a0a: link is not ready
Mar 15 02:44:29 rig01 dbus[951]: [system] Successfully activated service 'org.freedesktop.nm_dispatcher'
Mar 15 02:44:29 rig01 systemd[1]: Started Network Manager Script Dispatcher Service.
Mar 15 02:44:29 rig01 nm-dispatcher: req:1 'down' [wlx001ee5fc8a0a]: new request (1 scripts)
Mar 15 02:44:29 rig01 nm-dispatcher: req:1 'down' [wlx001ee5fc8a0a]: start running ordered scripts...
Mar 15 02:44:29 rig01 NetworkManager[972]: <info>  [1521042269.6235] dhcp4 (wlx001ee5fc8a0a): canceled DHCP transaction, DHCP client pid 1230
Mar 15 02:44:29 rig01 NetworkManager[972]: <info>  [1521042269.6236] dhcp4 (wlx001ee5fc8a0a): state changed bound -> done
Mar 15 02:44:29 rig01 NetworkManager[972]: <info>  [1521042269.6238] dhcp6 (wlx001ee5fc8a0a): canceled DHCP transaction
Mar 15 02:44:29 rig01 NetworkManager[972]: <info>  [1521042269.6245] dns-mgr: Writing DNS information to /sbin/resolvconf
Mar 15 02:44:29 rig01 dnsmasq[1240]: setting upstream servers from DBus
Mar 15 02:44:29 rig01 dnsmasq[1240]: using nameserver fe80::e2b9:e5ff:fe99:242%wlx001ee5fc8a0a#53
Mar 15 02:44:29 rig01 avahi-daemon[925]: Withdrawing address record for 10.1.1.51 on wlx001ee5fc8a0a.
Mar 15 02:44:29 rig01 avahi-daemon[925]: Leaving mDNS multicast group on interface wlx001ee5fc8a0a.IPv4 with address 10.1.1.51.
Mar 15 02:44:29 rig01 avahi-daemon[925]: Interface wlx001ee5fc8a0a.IPv4 no longer relevant for mDNS.
Mar 15 02:44:29 rig01 NetworkManager[972]: <info>  [1521042269.6297] dns-mgr: Writing DNS information to /sbin/resolvconf
Mar 15 02:44:29 rig01 dnsmasq[1240]: setting upstream servers from DBus
Mar 15 02:44:29 rig01 wpa_supplicant[1112]: wlx001ee5fc8a0a: Reject scan trigger since one is already pending
Mar 15 02:44:30 rig01 NetworkManager[972]: <info>  [1521042270.1772] device (wlx001ee5fc8a0a): supplicant interface state: scanning -> inactive
Mar 15 02:44:34 rig01 whoopsie[1121]: [02:44:34] Cannot reach: https://daisy.ubuntu.com
Mar 15 02:44:34 rig01 whoopsie[1121]: [02:44:34] Cannot reach: https://daisy.ubuntu.com
Mar 15 02:46:10 rig01 NetworkManager[972]: <info>  [1521042370.8421] policy: auto-activating connection 'iiNet990243'
Mar 15 02:46:10 rig01 NetworkManager[972]: <info>  [1521042370.8447] device (wlx001ee5fc8a0a): Activation: starting connection 'iiNet990243' (73f97426-79d7-4f8c-808b-341b37d71edd)
Mar 15 02:46:10 rig01 NetworkManager[972]: <info>  [1521042370.8452] device (wlx001ee5fc8a0a): state change: disconnected -> prepare (reason 'none') [30 40 0]
Mar 15 02:46:10 rig01 NetworkManager[972]: <info>  [1521042370.8453] manager: NetworkManager state is now CONNECTING
Mar 15 02:46:10 rig01 NetworkManager[972]: <info>  [1521042370.8457] device (wlx001ee5fc8a0a): state change: prepare -> config (reason 'none') [40 50 0]
Mar 15 02:46:10 rig01 NetworkManager[972]: <info>  [1521042370.8458] device (wlx001ee5fc8a0a): Activation: (wifi) access point 'iiNet990243' has security, but secrets are required.
Mar 15 02:46:10 rig01 NetworkManager[972]: <info>  [1521042370.8459] device (wlx001ee5fc8a0a): state change: config -> need-auth (reason 'none') [50 60 0]
Mar 15 02:46:10 rig01 NetworkManager[972]: <info>  [1521042370.8480] device (wlx001ee5fc8a0a): state change: need-auth -> prepare (reason 'none') [60 40 0]
Mar 15 02:46:10 rig01 NetworkManager[972]: <info>  [1521042370.8482] device (wlx001ee5fc8a0a): state change: prepare -> config (reason 'none') [40 50 0]
Mar 15 02:46:10 rig01 NetworkManager[972]: <info>  [1521042370.8484] device (wlx001ee5fc8a0a): Activation: (wifi) connection 'iiNet990243' has security, and secrets exist.  No new secrets needed.
Mar 15 02:46:10 rig01 NetworkManager[972]: <info>  [1521042370.8484] Config: added 'ssid' value 'iiNet990243'
Mar 15 02:46:10 rig01 NetworkManager[972]: <info>  [1521042370.8484] Config: added 'scan_ssid' value '1'
Mar 15 02:46:10 rig01 NetworkManager[972]: <info>  [1521042370.8485] Config: added 'key_mgmt' value 'WPA-PSK'
Mar 15 02:46:10 rig01 NetworkManager[972]: <info>  [1521042370.8485] Config: added 'auth_alg' value 'OPEN'
Mar 15 02:46:10 rig01 NetworkManager[972]: <info>  [1521042370.8485] Config: added 'psk' value '<omitted>'
...snip...
Mar 15 02:46:13 rig01 dhclient[12007]: DHCPREQUEST of 10.1.1.51 on wlx001ee5fc8a0a to 255.255.255.255 port 67 (xid=0x18a3e458)
Mar 15 02:46:18 rig01 dhclient[12007]: DHCPREQUEST of 10.1.1.51 on wlx001ee5fc8a0a to 255.255.255.255 port 67 (xid=0x18a3e458)
Mar 15 02:46:18 rig01 dhclient[12007]: DHCPACK of 10.1.1.51 from 10.1.1.1
Mar 15 02:46:18 rig01 NetworkManager[972]: <info>  [1521042378.4075]   address 10.1.1.51
Mar 15 02:46:18 rig01 NetworkManager[972]: <info>  [1521042378.4076]   plen 24 (255.255.255.0)
Mar 15 02:46:18 rig01 NetworkManager[972]: <info>  [1521042378.4076]   gateway 10.1.1.1
Mar 15 02:46:18 rig01 NetworkManager[972]: <info>  [1521042378.4076]   server identifier 10.1.1.1
Mar 15 02:46:18 rig01 NetworkManager[972]: <info>  [1521042378.4076]   lease time 86400
Mar 15 02:46:18 rig01 NetworkManager[972]: <info>  [1521042378.4076]   hostname 'rig01'
Mar 15 02:46:18 rig01 NetworkManager[972]: <info>  [1521042378.4076]   nameserver '10.1.1.1'
Mar 15 02:46:18 rig01 NetworkManager[972]: <info>  [1521042378.4076]   domain name 'lan'
```

Thus, what I think happened is that, the restart / reconnect somehow stopped the thread that reports the hashrate.

**Comment by jmsjr** [comment]
I guess what I am trying to say is that, the change to handle reconnects impacted the hashrate reporting.

**Comment by fsalle** [comment]
A simple way to reproduce it:
- Disable your DNS resolution
- In the logs you see loops of 'Could not resolve host eu1.ethermine.org:4444,  Host not found (non-authoritative), try again later'
- Reactivate your DNS
- Everything is fine but hashrate is not reported.

If it helps I also got (as it seems related):
_
Mar 15 09:07:29 miner ethminer:   ℹ  09:07:29|ethminer|  Shutting down...
Mar 15 09:07:29 miner ethminer:   ℹ  09:07:29|ethminer|  Disconnected from eu1.ethermine.org
Mar 15 09:07:29 miner ethminer:   ℹ  09:07:29|ethminer|  Shutting down miners...
Mar 15 09:07:29 miner ethminer:   ✘  09:07:29|stratum |  Read response failed: Operation canceled
Mar 15 09:07:29 miner ethminer: *** Error in `ethminer': double free or corruption (fasttop): 0x00007f44ec000a90 ***
Mar 15 09:07:29 miner ethminer: ethminer(_ZN16EthStratumClient10disconnectEv+0x1c8)[0x4e4cf8]
Mar 15 09:07:29 miner ethminer: ethminer(_ZN16EthStratumClient12readResponseERKN5boost6system10error_codeEm+0x2c5)[0x4efd45]
Mar 15 09:07:29 miner ethminer: ethminer(_ZN5boost4asio6detail26read_until_delim_string_opINS0_19basic_stream_socketINS0_2ip3tcpEEENS0_19basic_stream
buf_refISaIcEEENS_3_bi6bind_tIvNS_4_mfi3mf2Iv16EthStratumClientRKNS_6system10error_codeEmEENSA_5list3INSA_5valueIPSE_EEPFNS_3argILi1EEEvEPFNSO_ILi2EE
EvEEEEEEclESI_mi+0x249)[0x4e77d9]
Mar 15 09:07:29 miner ethminer: ethminer(_ZN5boost4asio6detail23reactive_socket_recv_opINS0_17mutable_buffers_1ENS1_26read_until_delim_string_opINS0_
19basic_stream_socketINS0_2ip3tcpEEENS0_19basic_streambuf_refISaIcEEENS_3_bi6bind_tIvNS_4_mfi3mf2Iv16EthStratumClientRKNS_6system10error_codeEmEENSC_
5list3INSC_5valueIPSG_EEPFNS_3argILi1EEEvEPFNSQ_ILi2EEEvEEEEEEEE11do_completeEPvPNS1_19scheduler_operationESK_m+0x121)[0x4e7e81]
Mar 15 09:07:29 miner ethminer: ethminer(_ZN5boost4asio6detail9scheduler3runERNS_6system10error_codeE+0x477)[0x4a3f47]
Mar 15 09:07:29 miner ethminer: ethminer(_ZN5boost4asio10io_context3runEv+0x36)[0x4a67a6]
Mar 15 09:07:29 miner ethminer: 00400000-008de000 r-xp 00000000 fc:00 1049637                            /usr/local/bin/ethminer
Mar 15 09:07:29 miner ethminer: 00add000-00aff000 r--p 004dd000 fc:00 1049637                            /usr/local/bin/ethminer
Mar 15 09:07:29 miner ethminer: 00aff000-00b0e000 rw-p 004ff000 fc:00 1049637                            /usr/local/bin/ethminer
Mar 15 09:07:29 miner ethminer: 7f450306c000-7f45038ad000 r-xp 00000000 fc:00 1052038                    /usr/lib/x86_64-linux-gnu/libcuda.so.390.30
Mar 15 09:07:29 miner ethminer: 7f45038ad000-7f4503aad000 ---p 00841000 fc:00 1052038                    /usr/lib/x86_64-linux-gnu/libcuda.so.390.30
Mar 15 09:07:29 miner ethminer: 7f4503aad000-7f4503bfe000 rw-p 00841000 fc:00 1052038                    /usr/lib/x86_64-linux-gnu/libcuda.so.390.30
Mar 15 09:07:29 miner001 rc.local[684]:   m  09:07:29|ethminer|  Speed  22.56 Mh/s    gpu/0 22.56 79C 50% 120W  [A278+8:R0+0:F0] Time: 12:01
Mar 15 09:07:34 miner001 rc.local[684]:   m  09:07:34|ethminer|  Speed  22.69 Mh/s    gpu/0 22.69 79C 50% 119W  [A278+8:R0+0:F0] Time: 12:02
Mar 15 09:07:37 miner ethminer:   m  09:07:37|ethminer|  ethminer version 0.14.0.dev4
Mar 15 09:07:37 miner ethminer:   m  09:07:37|ethminer|  Build: linux / release +git. 81ad571_

Thank you
Fred

PS: Ubuntu 16.04, trunk version

**Comment by kalltkaffe** [comment]
> I manually changed the code back, recompiled and then restarted mining. Time will tell if it helped.

24 hours later and I now have a full green line in the stats.
![image](https://user-images.githubusercontent.com/691687/37489753-f0fffc3a-2898-11e8-993d-5c48f2cd516c.png)

**Comment by vatanuki** [comment]
what about this patch?


[hashrate.diff.txt](https://github.com/ethereum-mining/ethminer/files/1817698/hashrate.diff.txt)

**Comment by kalltkaffe** [comment]
> what about this patch?

I'll give it a try now after successfully running fine 30+ hours with the changes in https://github.com/ethereum-mining/ethminer/commit/209254b6990119853d14d517bb3b315813cab062 reverted. I'll report back how it goes.

**Comment by kalltkaffe** [comment]
> >what about this patch?

>I'll give it a try now after successfully running fine 30+ hours with the changes in 209254b reverted. I'll report back how it goes.

Still works after 8 hours and 20 minutes but I think more time testing it is needed to be certain.

**Comment by vatanuki** [comment]
> Still works after 8 hours and 20 minutes but I think more time testing it is needed to be certain.

here is pull request https://github.com/ethereum-mining/ethminer/pull/897

**Comment by kalltkaffe** [comment]
> >Still works after 8 hours and 20 minutes but I think more time testing it is needed to be certain.

>here is pull request #897

Yes, I noticed that one. It's the same changes as the patch you posted here if I'm not mistaken. Been running 12 hours and 30 minutes now with those changes. Even pulled the network cable for 5 seconds as someone suggested. Still reports the hashrate just fine.

**Comment by kalltkaffe** [comment]
Close to 25 hours now. Still working fine. Thanks for the fix.

**Comment by jean-m-cyr** [comment]
Fixed by PR #897. Closing
